### DISCUSSION TITLE
Can I program maple mini with CubeMX IDE ?

### DISCUSSION THREAD

**Discussion by eagl1** [stm32duino/Arduino_Core_STM32#1611] [Q&A]
Hi,

I just installed CubeMX IDE on my computer, but I don't know if I can program a maple mini board and how.

1. Can you tell me if I can program the maple mini with CubeMX or not ?
2. If not, then what the best platform to program the maple mini with its default programming toolchain ?
3. I've programmed the maple mini with Arduino IDE but I want to run the original C lines of code. What is the best way to do so ?

Thanks,

**Comment by fpistm** [comment]
HI @eagl1 


> 1. Can you tell me if I can program the maple mini with CubeMX or not ?

No CubeMX allows to generate project for different target: STM32CubeIDE, IAR, ... 



> If not, then what the best platform to program the maple mini with its default programming toolchain ?

Several exists. You can refers to https://www.st.com/en/development-tools/stm32-ides.html



> I've programmed the maple mini with Arduino IDE but I want to run the original C lines of code. What is the best way to do so ?

I don't understand this point so can't help. What did you call "original C lines of code"

**Reply by eagl1** [reply]
> I don't understand this point so can't help. What did you call "original C lines of code"

I mean like the C core libraries used in the Arduino Hardware folder.
Which is this one:
[https://github.com/rogerclarkmelbourne/Arduino_STM32](url)

**Reply by eagl1** [reply]
I have this board:
[https://sparklogic.ru/wp-content/uploads/2019/11/img_5de0f6ad10544.jpg](https://sparklogic.ru/wp-content/uploads/2019/11/img_5de0f6ad10544.jpg)

How to program it ?

**Reply by fpistm** [reply]
> 

OK. So this core is not related to to this organization.
So the correct way to get support with this core is here: https://github.com/rogerclarkmelbourne/Arduino_STM32
And I saw you already enter one: https://github.com/rogerclarkmelbourne/Arduino_STM32/issues/865

You can use an stlink or the builtin bootloader using UART (see https://www.st.com/content/ccc/resource/technical/document/application_note/b9/9b/16/3a/12/1e/40/0c/CD00167594.pdf/files/CD00167594.pdf/jcr:content/translations/en.CD00167594.pdf)

**Reply by eagl1** [reply]
> You can use an stlink or the builtin bootloader using UART

I downloaded the document, but it's really complicated for me as I don't have any idea about advanced stuff like this board.

I can work the with maple mini with Arduino IDE, but the cz ministm32f103zx seem to be really difficult and I didn't understand what to do exactly to enable the bootloader.

Also when I connect the cz ministm32f103zx to the PC with the USB, it give me an error message that it's not recognized.

When I check for it through the device manager, it is displayed like this:

![image](https://user-images.githubusercontent.com/14331173/149773179-51a8f2d1-c2ba-4fe1-a545-dbd77aff3eba.png)


What does that mean ?

**Reply by fpistm** [reply]
It is not related to this core. 
About the message displayed it seems that the boards have a dedicated FTDI or something like this which require a driver. a simple google query give me this: http://www.prolific.com.tw/US/ShowProduct.aspx?p_id=225&pcid=41


For further question use the https://github.com/rogerclarkmelbourne/Arduino_STM32 or the forum in the LibMaple section.
